CMakeLists.txt

```diff
@@ -320,6 +320,13 @@ if(GINKGO_BUILD_DOC)
     add_subdirectory(doc)
 endif()
 
+
+# add escape character '\' and convert the list to string
+string(REPLACE " " "\ " PKG_CMAKE_INSTALL_PREFIX "${CMAKE_INSTALL_PREFIX}")
+list(TRANSFORM GINKGO_INTERFACE_LINK_FLAGS REPLACE " " "\\\\ ")
+list(TRANSFORM GINKGO_INTERFACE_CXX_FLAGS REPLACE " " "\\\\ ")
+string(REPLACE ";" " " GINKGO_INTERFACE_CXX_FLAGS "${GINKGO_INTERFACE_CXX_FLAGS}")
+string(REPLACE ";" " " GINKGO_INTERFACE_LINK_FLAGS "${GINKGO_INTERFACE_LINK_FLAGS}")
```

Comment:
Can you please elaborate a bit more on what this transformation is intended to do? Also it's not entirely obvious why it's necessary.

Reply:
the first three commands change the space ` ` to `\ `. string replace is plain text replacement such that it can use `\ `
However, list transform replace is regex replacement such that I need to put double escape. (`\\\\ `(escape \ for regex) -> `\\ ` (escape in the replacement) -> `\ ` in the final stage)
the last two are moved from inside to here.
I will also add the description into the file later

Reply:
So you escape spaces instead of putting them in quotes?

Reply:
yes

Reply:
and why not leave them in quotes instead? 😄